File: src/main/ts/rusk/resource/system/SystemResource.ts
module rusk {
    export module resource {
        export module system {
            export class SystemResource {
                private $http : ng.IHttpService;
                
                constructor($http : ng.IHttpService) {
                    this.$http = $http;
                }
                
                initialize() : void {
                    this.$http
                        .post('rest/system', {})
                        .success(() => {
                            console.log('success');
                        });
                }
            }
        }
    }
}

angular
.module('rusk')
.factory('systemResource', [
    '$http',
    ($http) => {
        return new rusk.resource.system.SystemResource($http);
    }
]);

File: src/main/java/rusk/rest/system/SystemResource.java
package rusk.rest.system;

import javax.inject.Inject;
import javax.ws.rs.POST;
import javax.ws.rs.Path;
import javax.ws.rs.core.Response;

import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

import rusk.system.db.DatabaseMigration;
import rusk.system.db.RuskConnectionPool;

/**
 * Rusk の初期化処理を提供するリソースクラス。
 */
@Path("system")
public class SystemResource {
    private static final Logger logger = LoggerFactory.getLogger(SystemResource.class);
    
    private final DatabaseMigration dbMigration;
    private final RuskConnectionPool ruskConnectionPool;
    
    @Inject
    public SystemResource(DatabaseMigration dbMigration, RuskConnectionPool ruskConnectionPool) {
        this.dbMigration = dbMigration;
        this.ruskConnectionPool = ruskConnectionPool;
    }

    @POST
    public Response initialize() {
        logger.debug("initialize system");
        
        this.dbMigration.migrate();
        this.ruskConnectionPool.initialize();
        
        return Response.status(Response.Status.OK).build();
    }
}
